openapi/mft/utils.bal

```diff
@@ -0,0 +1,277 @@
+            } else if (value is SimpleBasicType[]) {
+                param.push(getSerializedArray(key, value, encodingData.style, encodingData.explode));
+            } else if (value is record {}) {
+                if (encodingData.style == DEEPOBJECT) {
+                    param.push(getDeepObjectStyleRequest(key, value));
+                } else {
+                    param.push(getFormStyleRequest(key, value, encodingData.explode));
+                }
+            } else {
+                param.push(key, "=", value.toString());
+            }
+            param.push("&");
+        }
+        _ = param.pop();
+    }
+    string restOfPath = string:'join("", ...param);
+    return restOfPath;
+}
+
+# Generate header map for given header values.
+#
+# + headerParam - Headers  map
+# + return - Returns generated map or error at failure of client initialization
+isolated function getMapForHeaders(map<any> headerParam) returns map<string|string[]> {
+    map<string|string[]> headerMap = {};
+    foreach var [key, value] in headerParam.entries() {
+        if value is string || value is string[] {
+            headerMap[key] = value;
+        } else if value is int[] {
+            string[] stringArray = [];
+            foreach int intValue in value {
+                stringArray.push(intValue.toString());
+            }
+            headerMap[key] = stringArray;
+        } else if value is SimpleBasicType {
+            headerMap[key] = value.toString();
+        }
+    }
+    return headerMap;
+}
+
+isolated function createBodyParts(record {|anydata...; |} anyRecord, map<Encoding> encodingMap = {})
+returns mime:Entity[]|error {
+    mime:Entity[] entities = [];
+    foreach [string, anydata] [key, value] in anyRecord.entries() {
+        Encoding encodingData = encodingMap.hasKey(key) ? encodingMap.get(key) : {};
+        mime:Entity entity = new mime:Entity();
+        if value is byte[] {
+            entity.setByteArray(value);
+        } else if value is SimpleBasicType|SimpleBasicType[] {
+            entity.setText(value.toString());
+        } else if value is record {}|record {}[] {
+            entity.setJson(value.toJson());
+        }
+        if (encodingData?.contentType is string) {
+            check entity.setContentType(encodingData?.contentType.toString());
+        }
+        map<any>? headers = encodingData?.headers;
+        if (headers is map<any>) {
+            foreach var [headerName, headerValue] in headers.entries() {
+                if headerValue is SimpleBasicType {
+                    entity.setHeader(headerName, headerValue.toString());
+                }
+            }
+        }
+        entities.push(entity);
+    }
+    return entities;
+}
+
+# Encoding AS2identifier
+#
+# + as2Identifier - unencoded as2Identifier (e.g: <333900490041112@mftgateway.com> )
+# + return - Encoded string or error
+isolated function encodePathParam(string as2Identifier) returns string|error {
```

Comment:
Cant we reuse  `getEncodedUri` function since it does the same thing.

Reply:
Updated with changes